### ISSUE TITLE
Completion in doc comments eats the space when typing "code"

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#61223]
**Version Used**: Version 17.3.0 Preview 2.0 [32509.446.main]

**Steps to Reproduce**:

1. Create a doc comment.
2. Type `code` in the doc comment.
3. Hit space.

**Expected Behavior**:

Completion window is dismissed and the space is inserted.

**Actual Behavior**:

Completion window is dismissed and the space is not inserted.

**Comment by CyrusNajmabadi** [comment]
Note: this means typing `<code` then `<space>`.  Looks not limited to `code`.  Likely that `<space>` is a commit character.  Not sure what is causing the actual space to not get inserted though (as that's what we do for completion in non-doc-comment contexts).

**Comment by 333fred** [comment]
> Note: this means typing `<code` then `space`

No. This was very specifically _not_ starting an xml tag. I may have hit backspace as part of this, which causes the completion window to open.

**Comment by 333fred** [comment]
@CyrusNajmabadi and I looked offline. You need the "Show completion list after a character is deleted" setting on to reproduce this.

**Comment by CyrusNajmabadi** [comment]
I repro this as well.  Def a bug.  Even if that option is on, we should not be bringing up the completion list here as the user is not in an appropriate context for that identifier to be completed.
